cassandra-diagnostics-commons/src/main/java/io/smartcat/cassandra/diagnostics/module/Module.java

```diff
@@ -0,0 +1,40 @@
+package io.smartcat.cassandra.diagnostics.module;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.smartcat.cassandra.diagnostics.Query;
+import io.smartcat.cassandra.diagnostics.reporter.Reporter;
+
+/**
+ * Module abstraction that forces a constructor signature. All valid modules should extend this class.
+ */
+public abstract class Module {
+
+    /**
+     * Module configuration.
+     */
+    protected ModuleConfiguration configuration;
+
+    /**
+     * Reporter list defined in configuration.
+     */
+    public List<Reporter> reporters = new ArrayList<>();
+
+    /**
+     * Constructor.
+     *
+     * @param configuration Module configuration
+     */
+    public Module(ModuleConfiguration configuration) {
```

Comment:
The way ModuleConfiguration is now implies that a concrete Module implementation would be responsible for Reporters initialization. Or, at least, it is not clear how the Module instances would obtain references to Reporters (List<Reporter> reporters). On the other hand, as this is a quite generic thing and not dependent on specific Module implementation it should be addressed by this abstract class.